### PR TITLE
add user's claims to azure signalr access token

### PR DESCRIPTION
**Pull request by realwanpengli**
add user's claims to azure signalr access token

### CODE REVIEW

**Comment by realwanpengli** [comment]
auth sample will be update in next pr

File: src/SignalRServiceExtension/Client/AzureSignalRClient.cs [hunk 1]
```diff
@@ -26,12 +34,22 @@ internal AzureSignalRClient(IServiceHubContextStore serviceHubContextStore, ISer
             this.serviceManager = serviceManager;
         }
 
-        internal SignalRConnectionInfo GetClientConnectionInfo(string hubName, string userId)
+        internal SignalRConnectionInfo GetClientConnectionInfo(string hubName, string userId, string idToken, string[] claimTypeList)
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: Add some UT to cover the changes.

File: src/SignalRServiceExtension/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class Constants
+    {
+        public const string AzureSignalRUserPrefix = "asrs.u.";
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Constants.cs]: Either make this a private member in `AzureSignalRClient` or move some other const string to this file together. It seems not necessary to put it in constants alone.

File: src/SignalRServiceExtension/Client/AzureSignalRClient.cs [hunk 3]
```diff
@@ -26,12 +34,22 @@ internal AzureSignalRClient(IServiceHubContextStore serviceHubContextStore, ISer
...
         {
+            IEnumerable<Claim> selectedClaims = null;
+            if (idToken != null && claimTypeList != null && claimTypeList.Length > 0)
+            {
+                var jwtToken = new JwtSecurityTokenHandler().ReadJwtToken(idToken);
+                selectedClaims = from claim in jwtToken.Claims
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: Better to uniform the name, customerClaims | selectedClaims.

File: src/SignalRServiceExtension/Client/AzureSignalRClient.cs [hunk 4]
```diff
@@ -88,5 +106,25 @@ public async Task RemoveUserFromGroup(string hubName, string userId, string grou
             var serviceHubContext = await serviceHubContextStore.GetOrAddAsync(hubName);
             await serviceHubContext.UserGroups.RemoveFromGroupAsync(userId, groupName);
         }
+
+        private static IEnumerable<Claim> AddPrefixToDupClaimType(IEnumerable<Claim> customerClaims, string prefix)
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: Suggest to uniform the name with SDK side as BuildJwtClaims() would be better.

File: src/SignalRServiceExtension/Client/AzureSignalRClient.cs [hunk 5]
```diff
@@ -17,6 +18,14 @@ namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
 {
     internal class AzureSignalRClient : IAzureSignalRSender
     {
+        public const string AzureSignalRUserPrefix = "asrs.u.";
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: >public [](start = 8, length = 6)

private?

Comment (realwanpengli) [reply src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: It is used in UT

Comment (JialinXin) [reply src/SignalRServiceExtension/Client/AzureSignalRClient.cs]: OK.

**Pull request merged**